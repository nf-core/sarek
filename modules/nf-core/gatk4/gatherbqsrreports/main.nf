process GATK4_GATHERBQSRREPORTS {
    tag "$meta.id"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/gatk4:4.5.0.0--py36hdfd78af_0':
        'biocontainers/gatk4:4.5.0.0--py36hdfd78af_0' }"

    input:
    tuple val(meta), path(table)

    output:
    tuple val(meta), path("*.table"), emit: table
    path "versions.yml"             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def input_list = table.collect{"--input $it"}.join(' ')

    def avail_mem = 3072
    if (!task.memory) {
        log.info '[GATK GatherBQSRReports] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
    } else {
        avail_mem = (task.memory.mega*0.8).intValue()
    }
    """
    gatk --java-options "-Xmx${avail_mem}M -XX:-UsePerfData" \\
        GatherBQSRReports \\
        $input_list \\
        --output ${prefix}.table \\
        --tmp-dir . \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        gatk4: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$//')
    END_VERSIONS
    """
}
