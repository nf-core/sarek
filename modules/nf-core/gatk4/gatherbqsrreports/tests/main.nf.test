
nextflow_process {

    name "Test Process GATK4_GATHERBQSRREPORTS"
    script "../main.nf"
    process "GATK4_GATHERBQSRREPORTS"

    tag "modules"
    tag "modules_nfcore"
    tag "gatk4"
    tag "gatk4/gatherbqsrreports"

    test("test-gatk4-gatherbqsrreports") {

        when {
            process {
                """
                input[0] = [
				[ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/test.baserecalibrator.table', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-gatk4-gatherbqsrreports-multiple") {

        when {
            process {
                """
                input[0] = [
				[ id:'test', single_end:false ], // meta map
				    [
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/test.baserecalibrator.table', checkIfExists: true),
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/test2.baserecalibrator.table', checkIfExists: true)
                    ]
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
