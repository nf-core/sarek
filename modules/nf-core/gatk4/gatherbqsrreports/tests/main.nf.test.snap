{
    "test-gatk4-gatherbqsrreports-multiple": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.table:md5,0c1257eececf95db8ca378272d0f21f9"
                    ]
                ],
                "1": [
                    "versions.yml:md5,413fc0014d5dc41ab67d65f59f61a4a0"
                ],
                "table": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.table:md5,0c1257eececf95db8ca378272d0f21f9"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,413fc0014d5dc41ab67d65f59f61a4a0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T12:22:34.490694"
    },
    "test-gatk4-gatherbqsrreports": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.table:md5,9603b69fdc3b5090de2e0dd78bfcc4bf"
                    ]
                ],
                "1": [
                    "versions.yml:md5,413fc0014d5dc41ab67d65f59f61a4a0"
                ],
                "table": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.table:md5,9603b69fdc3b5090de2e0dd78bfcc4bf"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,413fc0014d5dc41ab67d65f59f61a4a0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-26T12:22:10.552951"
    }
}